### DISCUSSION TITLE
J2K transfer syntax support

### DISCUSSION THREAD

**Discussion by arunjoshtech** [Enet4/dicom-rs#412] [Q&A] [answered]
Does project support j2k transfer syntax?

**Comment by Enet4** [comment]
Yes, the `pixeldata` crate from the DICOM-rs project is at an early stage, but it supports JPEG 2000 encodings via GDCM bindings. You can try it out with the upstream revision. You can also keep track of development around pixeldata here: #125

**Comment by Enet4** [comment, marked as answer]
A heads-up, DICOM-rs 0.6.2 provides JPEG 2000 support via OpenJPEG. One can statically link to its reference implementation (`openjpeg-sys`) or use a C-to-Rust port (`openjp2`).

**Reply by bowenxuuu** [reply]
Hello Enet4,

I'm wondering which one to choose between `openjpeg-sys` and `openjp2`?  Could you give me some advice?

Thanks!

**Reply by Enet4** [reply]
This is documented at the [dicom-transfer-syntax-registry root](https://docs.rs/dicom-transfer-syntax-registry/latest/dicom_transfer_syntax_registry/), but in short, `openjpeg-sys` offers static linking to the official OpenJPEG implementation, whereas `openjp2` is a C-to-Rust port of OpenJPEG.
What I often do here is prefer `openjpeg-sys` for most platforms, and use `openjp2` when targeting WebAssembly.
